src/shogun/base/AnyParameter.h

```diff
@@ -168,6 +168,16 @@ namespace shogun
 			return m_init_function;
 		}
 
+		void add_callback_function(std::function<void()>&& method)
```

Comment:
could I suggest you carve out this callback stuff into another PR for improved review :)